packages/core/causalorder/testframework.go

```diff
@@ -55,67 +138,110 @@ func (t *TestFramework) EntityIDs(aliases ...string) (entityIDs []MockEntityID)
 	return
 }
 
-// region Mocks ////////////////////////////////////////////////////////////////////////////////////////////////////////
+// EvictEpoch evicts all Entities that are older than the given epoch.
+func (t *TestFramework) EvictEpoch(index epoch.Index) {
+	t.evictionManager.EvictEpoch(index)
+	t.CausalOrder.EvictEpoch(index)
+}
 
-type MockEntityID struct {
+// AssertOrdered asserts that the given Entities are ordered.
+func (t *TestFramework) AssertOrdered(aliases ...string) {
+	t.orderedEntitiesMutex.RLock()
+	defer t.orderedEntitiesMutex.RUnlock()
+
+	require.Equal(t.test, len(aliases), len(t.orderedEntities))
+
+	for _, alias := range aliases {
+		require.True(t.test, t.orderedEntities[alias])
+	}
+}
+
+// AssertEvicted asserts that the given Entities are evicted.
+func (t *TestFramework) AssertEvicted(aliases ...string) {
+	t.evictedEntitiesMutex.RLock()
+	defer t.evictedEntitiesMutex.RUnlock()
+
+	require.Equal(t.test, len(aliases), len(t.evictedEntities))
+
+	for _, alias := range aliases {
+		require.True(t.test, t.evictedEntities[alias])
+	}
```

Comment:
Would be nice to not evict from the entity provider (within the testframework) and retrieve the entity and check whether the corresponding flag is set. ie when not solid and evicted -> orphaned/invalid right? so then this check would validate in more depth and also whether the corresponding callback was executed correctly